app/controllers/PaymentController.scala

```diff
@@ -35,28 +36,49 @@ class PaymentController @Inject()(ws:WSClient,cc: ControllerComponents) extends
       "reference_id": "TICKET",
       "amount": {
         "currency_code": "GBP",
-        "value": "240.00"
+        "value": "69.00"
       }
     }
   ]
   }
   """)
     val token="Bearer "+getAccessToken()
-    val request=ws.url("https://api.sandbox.paypal.com/v2/checkout/orders").
+    Await.result(ws.url("https://api.sandbox.paypal.com/v2/checkout/orders").
       addHttpHeaders("Content-Type"->"application/json").
       addHttpHeaders("Authorization"->token).
-      post(json)
-      request//this should be removed
+      post(json),Duration(5,"seconds")).json
+
+
+
+
+
   }
-  def lol: Unit ={
-    val request=ws.url("https://api.sandbox.paypal.com/v2/payments/authorizations/")
+  def capturePayment(orderID:String)=Action{
+    val token="Bearer "+getAccessToken()
+//   val response=Await.result(ws.url("https://api.sandbox.paypal.com/v2/checkout/orders/"+orderID+"/authorize").
+//      addHttpHeaders("Content-Type"->"application/json").
+//      addHttpHeaders("Authorization"->token).get
+//      ,Duration(5,"seconds"))
+    val response="curl -v -X POST https://api.sandbox.paypal.com/v2/checkout/orders/"+orderID +"/capture -H \"Content-Type: application/json\" -H \"Authorization: "+token+"\""!!
+
+    Ok(response)
+
+
+
+
+
+
   }
 
   def index = Action {
-   // Ok(views.html.index(getAccessToken()))
-    Ok(createOrder().toString)
-   //Ok(getAccessToken())
-    Ok(views.html.payment(69.00f))
+    //Ok(createOrder().toString())
+    val json=createOrder()
+    val url=(json\"links"\1\\"href").last.toString.replace("\"","")
+    val orderID=(json\"id").get.toString.replace("\"","")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Usage of get on optional type.](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5261906)